integrations/google-vertex-ai.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: integration
+name: Google Vertex AI
+description: Use Google Vertex AI Models with Haystack
+authors:
+    - name: deepset
+      socials:
+        github: deepset-ai
+        twitter: deepset_ai
+        linkedin: deepset-ai
+pypi: https://pypi.org/project/google-vertex-haystack/
+repo: https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/google_vertex
+type: Model Provider
+report_issue: https://github.com/deepset-ai/haystack-core-integrations/issues
+logo: /logos/vertexai.png
+version: Haystack 2.0
+toc: true
+---
+
+### Table of Contents
+
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+    - [Gemini API models](#gemini-api-models)
+    - [PaLM API models](#palm-api-models)
+    - [Codey API models](#codey-api-models)
+    - [Imagen API models](#imagen-api-models)
+
+## Overview
+
+[Vertex AI](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/overview) is a machine learning (ML) platform that lets you train and deploy ML models and AI applications, and customize large language models (LLMs) for use in your AI-powered applications. This integration enables the usage of [models](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models) through Vertex AI API on Google Cloud Platform (GCP). 
+
+Haystack supports [Gemini API models](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models#gemini-models), [PaLM API models](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models#palm-models), [Codey APIs models](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models#codey-models), and [Imagen API models](https://cloud.google.com/vertex-ai/docs/generative-ai/learn/models#imagen-models) for task such as **text generation**, **function calling**, **image generation**, **visual question answering**, **code generation**, and **image captioning**.
```

Comment:
It would be great to have a list of the components that this integration introduces as a rule in these integration pages. And then we can link the documentation page when they're ready. Example like the Gradient usage section: https://haystack.deepset.ai/integrations/gradient (adding links to docs now)

That's my only comment 🎉